src/panel/PoiPanel.jsx

```diff
@@ -0,0 +1,237 @@
+    sourceCategory: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props);
+    this.poi = props.poi;
+    if (props.poi) {
+      this.poi = PoiStore.deserialize(props.poi);
+    }
+    this.state = {
+      isCollapsed: true,
+      card: true,
+    };
+    this.categories = CategoryService.getCategories();
+    this.isDirectionActive = nconf.get().direction.enabled;
+    this.isMasqEnabled = nconf.get().masq.enabled;
+
+    window.execOnMapLoaded(() => {
+      const elem = document.querySelector('.poi_panel__content__card');
+
+      if (!elem) {
+        return;
+      }
+      fire(
+        'move_mobile_bottom_ui',
+        elem.offsetHeight + 20
+      );
+    });
+  }
+
+  expandCollapse = () => {
+    this.setState(state => ({
+      isCollapsed: !state.isCollapsed,
+    }));
+  }
+
+  backToFavorite = () => {
+    Telemetry.add(Telemetry.POI_BACKTOFAVORITE);
+    window.app.navigateTo('/favs');
+  }
+
+  showDetail = () => {
+    if (this.poi.meta && this.poi.meta.source) {
+      Telemetry.add(Telemetry.POI_SEE_MORE, null, null,
+        Telemetry.buildInteractionData({
+          id: this.poi.id,
+          source: this.poi.meta.source,
+          template: 'single',
+          zone: 'detail',
+          element: 'more',
+        })
+      );
+    }
+    this.setState({
+      card: false,
+    });
+  }
+
+  center = () => {
+    if (this.poi.meta && this.poi.meta.source) {
+      Telemetry.add('go', 'poi', this.poi.meta.source);
+    }
+    fire('fit_map', this.poi, layouts.POI);
+  }
+
+  openShare = () => {
+    if (this.poi.meta && this.poi.meta.source) {
+      Telemetry.add('share', 'poi', this.poi.meta.source);
+    }
+    if (this.poi) {
+      openShareModal(this.poi.toAbsoluteUrl());
+    }
+  }
+
+  closeAction = () => {
+    window.app.navigateTo('/');
+  }
+
+  openCategory = category => {
+    window.app.navigateTo(`/places/?type=${category.name}`);
+  }
+
+  backToList = () => {
+    Telemetry.add(Telemetry.POI_BACKTOLIST);
+    fire('restore_location');
+    window.app.navigateTo(`/places/?type=${this.props.sourceCategory}`);
+  }
+
+  backToSmall = () => {
+    this.setState({
+      card: true,
+    });
+  }
+
+  render() {
+    if (!this.poi) {
+      return null;
+    }
+    const data = {
+      className: 'poi_panel__back_to_list',
+    };
+    const pagesjaunes = this.poi.isFromPagesjaunes && this.poi.isFromPagesjaunes() ?
+      <img className="poi_panel__back_to_list_logo"
+        src="./statics/images/pagesjaunes.svg"
+        alt="PagesJaunes" />
+      : null;
+
+    if (this.props.isFromFavorite) {
+      data.callback = this.backToFavorite;
+      data.text = _('Back to favorite');
+    } else if (this.props.isFromCategory) {
+      data.callback = this.backToList;
+      data.text = _('Back to list');
+    } else if (!this.state.card) {
+      data.callback = this.backToSmall;
+      data.text = _('Back');
+      data.className = 'poi_panel__back_mobile';
+    }
+    return <div>
```

Comment:
This should re-use the `<Panel>` component somewhere. I know things are complicated by the mobile display which uses a card, but otherwise that's totally what the `<Panel>` component has been created for. This way you don't need to care about many classnames, the close button, etc.

Reply:
Let's keep it for after this PR is merged